examples/IncludeDatum.purs

```diff
@@ -0,0 +1,121 @@
+  , textEnvelopeBytes
+  )
+import Contract.Transaction
+  ( TransactionHash
+  , awaitTxConfirmed
+  , balanceAndSignTxE
+  , lookupTxHash
+  , plutusV1Script
+  , submit
+  )
+import Contract.TxConstraints (TxConstraints)
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value as Value
+import Ctl.Internal.Plutus.Types.TransactionUnspentOutput (_input)
+import Data.Array (head)
+import Data.BigInt as BigInt
+import Data.Lens (view)
+import Data.Map as Map
+
+main :: Effect Unit
+main = example testnetNamiConfig
+
+example :: ConfigParams () -> Effect Unit
+example cfg = launchAff_ do
+  runContract cfg do
+    logInfo' "Running Examples.IncludeDatum"
+    validator <- includeDatumScript
+    let vhash = validatorHash validator
+    logInfo' "Attempt to lock value"
+    txId <- payToIncludeDatum vhash
+    awaitTxConfirmed txId
+    logInfo' "Tx submitted successfully, Try to spend locked values"
+    spendFromIncludeDatum vhash validator txId
+  publishTestFeedback true
+
+datum :: Datum
+datum = Datum $ Integer $ BigInt.fromInt 42
+
+payToIncludeDatum :: ValidatorHash -> Contract () TransactionHash
+payToIncludeDatum vhash = do
+  let
+    constraints :: TxConstraints Unit Unit
+    constraints =
+      ( Constraints.mustPayToScript vhash datum Constraints.DatumWitness
+          $ Value.lovelaceValueOf
+          $ BigInt.fromInt 2_000_000
+      )
+        <> Constraints.mustIncludeDatum datum
+
+    lookups :: Lookups.ScriptLookups PlutusData
+    lookups = mempty
+  buildBalanceSignAndSubmitTx lookups constraints
+
+spendFromIncludeDatum
+  :: ValidatorHash
+  -> Validator
+  -> TransactionHash
+  -> Contract () Unit
+spendFromIncludeDatum vhash validator txId = do
+  let scriptAddress = scriptHashAddress vhash
+  utxos <- fromMaybe Map.empty <$> utxosAt scriptAddress
+  case view _input <$> head (lookupTxHash txId utxos) of
+    Just txInput ->
+      let
+        lookups :: Lookups.ScriptLookups PlutusData
+        lookups = Lookups.validator validator
+          <> Lookups.unspentOutputs utxos
+
+        constraints :: TxConstraints Unit Unit
+        constraints =
+          Constraints.mustSpendScriptOutput txInput unitRedeemer
+            <> Constraints.mustIncludeDatum datum
+      in
+        do
+          spendTxId <- buildBalanceSignAndSubmitTx lookups constraints
+          awaitTxConfirmed spendTxId
+          logInfo' "Successfully spent locked values."
+    _ -> logInfo' "no locked output at address"
+
+buildBalanceSignAndSubmitTx
+  :: Lookups.ScriptLookups PlutusData
+  -> TxConstraints Unit Unit
+  -> Contract () TransactionHash
+buildBalanceSignAndSubmitTx lookups constraints = do
+  ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+  bsTx <- liftedE $ balanceAndSignTxE ubTx
+  txId <- submit bsTx
+  logInfo' $ "Tx ID: " <> show txId
+  pure txId
+
+foreign import includeDatum :: String
+
+includeDatumScript :: Contract () Validator
```

Comment:
Please add a comment about the script semantics to this function. Also, let's rename 
```suggestion
only42Script :: Contract () Validator
```

Reply:
saw that there is an example named redeemer1, so I renamed mine to datum42

Reply:
Let's only rename the script itself. Because the example name should be descriptive. Could you please revert